src/Symfony/Component/Form/Tests/Extension/Core/Type/ExtendedChoiceTypeTest.php

```diff
@@ -36,6 +38,44 @@ public function testChoicesAreOverridden($type)
         $this->assertSame('b', $choices[1]->value);
     }
 
+    /**
+     * @dataProvider provideTestedTypes
+     */
+    public function testChoicesAreOverridden($type)
+    {
+        $factory = Forms::createFormFactoryBuilder()
+            ->addTypeExtension(new ChoiceTypeExtension($type))
+            ->getFormFactory()
+        ;
+
+        $choices = $factory->create($type, null, array('choice_loader' => null))->createView()->vars['choices'];
+
+        $this->assertCount(2, $choices);
+        $this->assertSame('A', $choices[0]->label);
+        $this->assertSame('a', $choices[0]->value);
+        $this->assertSame('B', $choices[1]->label);
+        $this->assertSame('b', $choices[1]->value);
+    }
+
+    /**
+     * @dataProvider provideTestedTypes
+     */
+    public function testChoiceLoaderIsOverridden($type)
+    {
+        $factory = Forms::createFormFactoryBuilder()
+            ->addTypeExtension(new LazyChoiceTypeExtension($type))
```

Comment:
I would keep it like it for consistency with the test above, and because it test both overriding the option directly and doing it with an extension, so looks safer as is.